src/stcal/resample/resample.py

```diff
@@ -1014,80 +1017,42 @@ def resample_variance_arrays(self, model, pixmap, iscale,
-            else:
-                mask = np.full_like(rn_var, False)
-
-            # Set the weight for the image from the weight type
-            if self.weight_type.startswith("ivm") and rn_var is not None:
-                weight = np.ones(self.output_array_shape)
-                weight[mask] = np.reciprocal(rn_var[mask])
-
-            elif self.weight_type == "exptime":
-                t, _ = get_tmeasure(model)
-                weight = np.full(self.output_array_shape, t)
-
-            # Weight and add the readnoise variance
-            # Note: floating point overflow is an issue if variance weights
-            # are used - it can't be squared before multiplication
-            if rn_var is not None:
-                # Add the inverse of the resampled variance to a running sum.
-                # Update only pixels (in the running sum) with
-                # valid new values:
-                mask = (rn_var >= 0) & np.isfinite(rn_var) & (weight > 0)
-                self._var_rnoise_wsum[mask] = np.nansum(
-                    [
-                        self._var_rnoise_wsum[mask],
-                        rn_var[mask] * weight[mask] * weight[mask]
-                    ],
-                    axis=0
-                )
-                self._var_rnoise_weight[mask] += weight[mask]
-
-        # Now do poisson and flat variance, updating only valid new values
-        # (zero is a valid value; negative, inf, or NaN are not)
-        if self._check_var_array(model, "var_poisson"):
-            pn_var = self._resample_one_variance_array(
-                "var_poisson",
-                model=model,
-                **pars,
-            )
-            if pn_var is not None:
-                mask = (pn_var >= 0) & np.isfinite(pn_var) & (weight > 0)
-                self._var_poisson_wsum[mask] = np.nansum(
-                    [
-                        self._var_poisson_wsum[mask],
-                        pn_var[mask] * weight[mask] * weight[mask]
-                    ],
-                    axis=0
+        for varname in self._enable_var:
+            if self._check_var_array(model, varname):
+                var = self._resample_one_variance_array(
+                    varname,
+                    model=model,
+                    **pars,
                 )
-                self._var_poisson_weight[mask] += weight[mask]
+                self.variance_info[varname]['var'] = var
 
-        if self._check_var_array(model, "var_flat"):
-            flat_var = self._resample_one_variance_array(
-                "var_flat",
-                model=model,
-                **pars,
-            )
-            if flat_var is not None:
-                mask = (flat_var >= 0) & np.isfinite(flat_var) & (weight > 0)
-                self._var_flat_wsum[mask] = np.nansum(
+        # Set the weight for the image from the weight type
+        if self.weight_type.startswith("ivm"):
+            rn_var = self.variance_info['var_rnoise']['var']
```

Comment:
The downstream tests are failing because of these lines - the failing tests have empty or incorrectly sized variance arrays, which fail L1021's `self._check_var_array`, leading to no `var` key definition for the `var_rnoise` entry. Then L1031 attempts to access the `var` key.

Reply:
Right, I understand why they're failing.  My question is more whether the test we have there is a good test in the first place---in my opinion, if someone specifies inverse variance read noise weighting and then gives input models without read noise variances, it's appropriate to fail.  I can write the code to handle this case, but I thought I'd check that this was really something we want to be able to handle.

Reply:
Apologies, I read the message that tagged me in, but neglected to read the first comment. I agree that these tests are covering a case that should probably return an error - I'm not sure it's a case that we should expect to see with any regularity, and if so, that it should only return a warning.

This will require a small jwst PR to update.

Reply:
@mcara , do you want to weigh in on whether to remove the offending jwst test or update this PR to handle the case that ivm weighting is selected but the input models don't include read noise variances?

Reply:
With regard to this particular crash, would replacing `startswith` with strict equality solve the issue?

Reply:
I think the test is doing what it is supposed to do: check that the code behaves "correctly" when inputs are incorrect. The code on `main` in `stcal` is reproducing old/original behavior in `jwst` pipeline (and I assume in `romancal` as well): https://github.com/spacetelescope/jwst/blob/6902a68f15d0a762d30915830384f9fdc02e9271/jwst/resample/resample.py#L298-L304

I do not know why the code was written to behave this way (log a message and proceed) but I assume it is because pipeline is not supposed to crash (and I had to remove multiple `raise Exception` from my `tweakreg` and other code to make it not fail). Maybe this has changed but it is for @tapastro to make a decision.

Reply:
However, the code currently in `stcal` is also buggy as `weight` would not be defined if check `var_rnoise` fails but other arrays pass. I think the cleanest would be to skip resampling of variance arrays when `var_rnoise` is missing and `weight_type` is (or stars with) "ivm" (depending on whether a particular weighting needs `rnoise`).

Reply:
I fully support avoiding error returns whenever possible - our code is first and foremost meant to support automated operations which deliver science products, and graceful handling of unexpected circumstances (especially during operations on arrays that are not the flux array) is preferable. I would suggest that the code log a warning message if it is unable to proceed with variance array resampling in the manner it expects and continues.

I don't know if Roman has different expectations.

Reply:
Note that in this case the flux array is impacted---the user has asked for ivm resampling, but var_rnoise does not exist, and so the weights used for resampling end up being trivial.  But I'll tweak this to avoid the error.